hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/selective/SelectiveSupervisorController.java

```diff
@@ -0,0 +1,93 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.common.config.ConfigFactory;
+import pl.allegro.tech.hermes.common.metric.HermesMetrics;
+import pl.allegro.tech.hermes.consumers.subscription.cache.SubscriptionsCache;
+import pl.allegro.tech.hermes.consumers.supervisor.ConsumersSupervisor;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.SupervisorController;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.WorkTracker;
+
+import static java.util.Collections.emptyList;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_WORKLOAD_ALGORITHM;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_WORKLOAD_CONSUMERS_PER_SUBSCRIPTION;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_WORKLOAD_NODE_ID;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_WORKLOAD_MAX_SUBSCRIPTIONS_PER_CONSUMER;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_WORKLOAD_REBALANCE_INTERVAL;
+
+public class SelectiveSupervisorController implements SupervisorController {
+    private ConsumersSupervisor supervisor;
+    private SubscriptionsCache subscriptionsCache;
+    private WorkTracker workTracker;
+    private ConsumerNodesRegistry consumersRegistry;
+    private ConfigFactory configFactory;
+    private HermesMetrics metrics;
+
+    private static final Logger logger = LoggerFactory.getLogger(SelectiveSupervisorController.class);
+
+    public SelectiveSupervisorController(ConsumersSupervisor supervisor,
+                                         SubscriptionsCache subscriptionsCache,
+                                         WorkTracker workTracker,
+                                         ConsumerNodesRegistry consumersRegistry,
+                                         ConfigFactory configFactory,
+                                         HermesMetrics metrics) {
+
+        this.supervisor = supervisor;
+        this.subscriptionsCache = subscriptionsCache;
+        this.workTracker = workTracker;
+        this.consumersRegistry = consumersRegistry;
+        this.configFactory = configFactory;
+        this.metrics = metrics;
+    }
+
+    @Override
+    public void onSubscriptionAssigned(Subscription subscription) {
+        logger.info("Assigning consumer for {}", subscription.getId());
+        supervisor.assignConsumerForSubscription(subscription);
+    }
+
+    @Override
+    public void onAssignmentRemoved(SubscriptionName subscription) {
+        logger.info("Removing assignment from consumer for {}", subscription.getId());
+        supervisor.deleteConsumerForSubscriptionName(subscription);
+    }
+
+    @Override
+    public void start() throws Exception {
+        subscriptionsCache.start(emptyList());
+        workTracker.start(ImmutableList.of(this));
+        supervisor.start();
```

Comment:
Every consumers module listens for retransmission events, even for subscriptions for which is not assigned in selective algorithm. Retransmission spawns new consumer so you will have as many consumers for retransmitted subscription as number of consumers modules which is unexpected in selective algorithm.